cli/connect/interactive.go

```diff
@@ -0,0 +1,245 @@
+			formatterOpts.NoGraphics = false
+			log.Info("Pseudo graphics is enabled now")
+		} else if trimmed == setPseudoGraphicsDisable {
+			formatterOpts.NoGraphics = true
+			log.Info("Pseudo graphics is disabled now")
+		} else {
+			panic("there is no pattern cases for handling pseudo graphics enabling/disabling")
+		}
+	} else {
+		log.Error("Pseudo graphics enabling/disabling " +
+			"only allowed for table and ttable output formats")
+	}
+}
+
+// handleFormatOption handles output interactive option.
+func handleFormatOption(trimmed string, executorEval *string,
+	console *Console, formatterOpts *formatter.Opts) {
+	var newFormat string
+	if strings.HasPrefix(trimmed, setFormatPrefix) {
+		newFormat = strings.TrimPrefix(trimmed, setFormatPrefix)
+	}
+	if strings.HasPrefix(trimmed, setFormatShortPrefix) {
+		if trimmed == setFormatShortPrefix {
+			if console.outputFormat == formatter.DefaultFormat {
+				console.outputFormat = formatter.YamlFormat
+				formatterOpts.TransposeMode = formatter.IsTTableFormat(console.outputFormat)
+			}
+			console.outputFormat = 1 + (console.outputFormat % 4)
+			formatterOpts.TransposeMode = formatter.IsTTableFormat(console.outputFormat)
+			trimmed = console.outputFormat.String()
+		}
+		newFormat = strings.TrimPrefix(trimmed, setFormatShortPrefix)
+	}
+	if outputFormat, ok := formatter.ParseFormat(newFormat); ok {
+		if outputFormat == formatter.TableFormat || outputFormat == formatter.TTableFormat {
+			*executorEval = consoleEvalFuncBodyTblsFmt
+			console.outputFormat = outputFormat
+			formatterOpts.TransposeMode = formatter.IsTTableFormat(console.outputFormat)
+			log.Infof("Selected output format: %s", console.outputFormat.String())
+		} else {
+			*executorEval = consoleEvalFuncBody
+			console.outputFormat = outputFormat
+			formatterOpts.TransposeMode = formatter.IsTTableFormat(console.outputFormat)
+			log.Infof("Selected output format: %s", console.outputFormat.String())
+		}
+	} else {
+		log.Warnf("Unsupported output format: %s", newFormat)
+	}
+}
+
+// handleTableDialectOption handles table dialect interactive option
+// for table/ttable output formats.
+func handleTableDialectOption(trimmed string, formatterOpts *formatter.Opts) {
+	newTableDialect := strings.TrimPrefix(trimmed, setTableDialectPrefix)
+	if tableDialect, ok := formatter.ParseTableDialect(newTableDialect); ok {
+		formatterOpts.TableDialect = tableDialect
+		if formatterOpts.TableDialect != formatter.DefaultTableDialect &&
+			formatterOpts.ColWidthMax > 0 {
+			formatterOpts.ColWidthMax = 0
+			log.Info("Selected max width: disabled")
+		}
+		log.Infof("Selected table dialect: %s", formatterOpts.TableDialect.String())
+	} else {
+		log.Warnf("Unsupported table dialect: %s", newTableDialect)
+	}
+}
+
+// handleInteractiveOption handles slash options for interactive console and returns
+// true if slash options detected in user input.
+func handleInteractiveOption(in string, executorEval *string,
```

Comment:
This code can be organized more simply. The idea:

1. Create a map: `option(s)_name[handle_function]`.
2.
```Go
for option, fun := range option_map {
     if strings.HasPrefix(trimmed, option) {
           fun(trimmied, console, opts)
     }
}
```

Reply:
Hmm, unfortunately, the detection of several options is not performed only through `HasPrefix`, plus, some options have a check for emptiness. Maybe it's worth to split `handleInteractiveOption` to smaller functions?

Reply:
> Hmm, unfortunately, the detection of several options is not performed only through `HasPrefix`

As example?

Reply:
Done

Reply:
Why do we could not replace the slice `interactiveOptsCases` to a map of [prefx]handleFunction ?

Reply:
The detecting logic of particular option may differ from others:

* Checking of option argument presence
* Validation of option argument
* Detecting can be based in `trimmed` or `in` value from user
* Checking the status of other options

So, I have to move this logic to `detectInteractiveOption` functions

Reply:
All detect* calls begins with:

```Go
if opts.trimmed == 'some' {
}
```
or
```Go
if strings.HasPrefix(opts.trimmed, 'some') {
}
```

It could be possible to use the map for it:

```Go
for opt, fun := range map {
    if strings.HasPrefix(trimmed, opt) {
         fun(data_without_trimmed)
    }
}
```

Reply:
Sorry, but not all:
* `detectHelperCase`
* `detectTableGraphicsCase`
* `detectOutputFormatCase`
* `detectTableDialectCase`

And some have some preprocessing logic (described above)

Could we use current implementation? The current solution is extensible via `[...]detectInteractiveOption` and flexible in terms of adding custom logic for processing options via `type detectInteractiveOption`

Reply:
The current approach certainly works, but it is not extensible in any way (for example, if we need to add autocompletion). It's hard to maintain and easy to make mistakes.

In addition, you are trying to implement different logic for the long and short forms in the same handler. This complicates the logic of potentially simple functions.

```Go
type optHandler struct { // or even an interface
    handler                func(postfix string, opts handlerOpts) error
    // autocompleter  func(postfix string, opts handlerOpts) ([]string, error)
}

commands := map[string]optHandler {
    "\\help": helpHandler,
    "\\":        helpHandler,
    "\\?":      helpHandler,
    setPseudoGraphicsEnable: handlePseudoGraphicsEnable,
    setPseudoGraphicsDisable: handlePseudoGraphicsDisable,
    setFormatPrefix: handleFormatLong,
    setFormatShortPrefix: handleFormatShort,
    setTableDialectPrefix: handleTableDialect,
}
```

It will be much easier to understand and maintain the list of options with such mapping. In addition, you won't need to manually strip the prefix/postfix in every function. Just check it and implement the logic of a handler.

Reply:
Done